src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeProvider.cs

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Resources;
+using Bicep.Core.TypeSystem.Types;
+
+namespace Bicep.Core.TypeSystem.Providers.ThirdParty
+{
+    public class ThirdPartyResourceTypeProvider : ResourceTypeProviderBase, IResourceTypeProvider
+    {
+        public const string NamePropertyName = "name";
+        public const string MetadataPropertyName = "metadata";
+        public const string NamespaceProperty = "namespace";
```

Comment:
These are all k8s specific - I'm guessing they were copied from the k8s provider.